lib/modules/platform/github/index.ts

```diff
@@ -1494,7 +1494,10 @@ export async function createPr({
   // istanbul ignore if
   if (config.forkToken) {
     options.token = config.forkToken;
-    options.body.maintainer_can_modify = true;
```

Comment:
It had been my assumption until now that this feature was quite useful for users. Therefore I want to think whether this change is necessary or if there's "another way".

Reply:
yeah, it totally is. I think in the issue I was advocating for a simpler branch deletion policy, like:

* If a PR is for the onboarding branch, never delete it
* If a PR is modified
   * only delete if there is no associated PR

but it seemed like changing the process here would be a rather larger undertaking, while for forked PRs this is already an improvement compared to renovate suddenly not updating a certain dependency at all anymore

Reply:
this is a breaking change, so it should better be configurable and enabled by default

Reply:
yeah, I can make it configurable. Though could also see it as a fix because the current "intended behavior" does not work and has more significant consequences (like people not knowing that they miss updates after having edited a fork branch)

Reply:
doesn't the delete branch after merge function (GitHub repo settings) solve the issue too?

Reply:
Interesting idea! (didn't know this existed). It looks like it wouldn't be able to delete branches from the fork account though. e.g. consider https://github.com/angular/dev-infra/pull/821 where the PR branch is not an upstream branch
<img width="778" alt="Screenshot 2023-01-10 at 21 23 35" src="https://user-images.githubusercontent.com/4987015/211654685-2e49f8e8-887f-4a94-9de5-06e813c903a2.png">

We had the setting enabled but it didn't delete the branch. Likely a maintainer cannot delete a branch from a user fork

Reply:
I agree with https://github.com/renovatebot/renovate/pull/19771#discussion_r1066218416